source/heat/heat_transfer_problem.cpp

```diff
@@ -29,10 +29,15 @@ namespace MeltPoolDG::Heat
         scratch_data->get_pcout() << "t= " << std::setw(10) << std::left
                                   << time_iterator.get_current_time();
 
-        if (base_in->parameters.heat.velocity != 0.0)
-          compute_field_vector(velocity,
-                               velocity_dof_idx,
-                               *base_in->get_velocity_field("heat_transfer"));
+        const auto source_field_function = base_in->get_source_field("heat_transfer");
+        if (source_field_function)
```

Comment:
I am not entirely sure but couldn't you simply write
```suggestion
        if (const auto source_field_function = base_in->get_source_field("heat_transfer"))
```